smdebug/tensorflow/collection.py

```diff
@@ -81,7 +81,7 @@ def add_for_mode(self, arg, mode=None):
             return self.add_variable(arg, mode=mode)
         elif isinstance(arg, tf.Tensor):
             return self.add_tensor(arg, mode=mode)
-        elif isinstance(arg, values.MirroredVariable):
+        elif isinstance(arg, (values.MirroredVariable, values.DistributedVariable)):
```

Comment:
both MirroredVariable and SyncOnReadVariable are of type values.DistributedVariable. Probably could just check for values.DistributedVariable

Reply:
If both values.MirroredVariable and SyncOnReadVariable are of the type, then it probably makes sense to type check only the parent.

Also can you rename the `self.add_mirrored_variable`

Reply:
done in the latest commit